src/synapse/azext_synapse/custom.py

```diff
@@ -121,59 +123,126 @@ def create_workspace(cmd, client, resource_group_name, workspace_name, account_u
+    # get the location of the workspace
+    workspace_client = cf_synapse_client_workspace_factory(cmd.cli_ctx)
+    workspace_object = workspace_client.get(resource_group_name, workspace_name)
+    location = workspace_object.location
+
     big_data_pool_info = BigDataPoolResourceInfo(location=location, spark_version=spark_version, node_size=node_size,
                                                  node_count=node_count, node_size_family=node_size_family,
                                                  spark_events_folder=spark_events_folder,
                                                  default_spark_log_folder=default_spark_log_folder, tags=tags)
-    if auto_scale_enabled:
-        big_data_pool_info.auto_scale = AutoScaleProperties(enabled=auto_scale_enabled, min_node_count=min_node_count,
-                                                            max_node_count=max_node_count)
-    if auto_pause_enabled:
-        big_data_pool_info.auto_pause = AutoPauseProperties(enabled=auto_pause_enabled,
-                                                            delay_in_minutes=delay_in_minutes)
-
-    if library_requirements_filename or library_requirements_content:
-        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_filename,
+
+    big_data_pool_info.auto_scale = AutoScaleProperties(enabled=enable_auto_scale, min_node_count=min_node_count,
+                                                        max_node_count=max_node_count)
+
+    big_data_pool_info.auto_pause = AutoPauseProperties(enabled=enable_auto_pause,
+                                                        delay_in_minutes=delay_in_minutes)
+
+    if library_requirements_file:
+        library_requirements_content = read_file_content(library_requirements_file)
+        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_file,
                                                                       content=library_requirements_content)
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, workspace_name, spark_pool_name,
                        big_data_pool_info, force)
 
 
-def update_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, tags=None):
-    return client.update(resource_group_name, workspace_name, spark_pool_name, tags)
+def update_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name,
+                      node_size=None, node_count=None, enable_auto_scale=None,
+                      min_node_count=None, max_node_count=None,
+                      enable_auto_pause=None, delay_in_minutes=None,
+                      library_requirements_file=None, tags=None, force=False, no_wait=False):
+    existing_spark_pool = client.get(resource_group_name, workspace_name, spark_pool_name)
+
+    if not existing_spark_pool:
+        raise CLIError("Failed to discover the spark pool {}".format(spark_pool_name))

```

Comment:
If spark pool doesn't exisit,  will the `client.get` method throw exception or just return the null value?

Reply:
it throws exception. I have removed this logic.

---
In reply to: [417127003](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r417127003) [](ancestors = 417127003)